action.yml

```diff
@@ -98,36 +98,35 @@ runs:
 
     - name: Build dependencies.log (and commit)
       run: |
-        pip3 install pip-tools
-
-        # do main dependencies.log in subproc
-        echo
-        echo "dependencies.log"
-        git mv requirements.txt dependencies.log || true
-        echo "pip-compile..."
-        pip-compile --upgrade --output-file="dependencies.log" &
-
-        # get all extras
-        EXTRAS=$(python3 ${{ github.action_path }}/list_extras.py setup.cfg)
-
-        # generate dependencies-*.log for each extras_require (each in a subproc)
-        for extra in $EXTRAS; do
-          echo
-          extra_dep_log="dependencies-${extra}.log"
-          echo $extra_dep_log
-          git mv "requirements-${extra}.txt" $extra_dep_log || true
-          echo "pip-compile..."
-          pip-compile --upgrade --extra $extra --output-file="$extra_dep_log" &
-        done
-        echo
+        # append permissive line to .gitignore since *.log is commonly present
+        line='!dependencies*.log'
+        if [[ ! $(grep -F "$line" .gitignore) ]]; then
+          echo "" >> .gitignore
+          echo "" >> .gitignore
+          echo "# wipac-dev-py-setup-action" >> .gitignore
+          echo "$line" >> .gitignore
+          git add .gitignore
+          git commit -m "<bot> update .gitignore" || true  # fails if no change
+          tail .gitignore
+        fi
 
-        # wait for all subprocs
-        wait -n # main dependencies.log
-        for extra in $EXTRAS; do
-          wait -n
-        done
+        # remove any old ones, then regenerate only what's needed
+        rm dependencies*.log
+
+        # Build
+        if [ -f ./Dockerfile ]; then
+          # from docker...
+          for f in ./Dockerfile*; do
+              echo $f
+              ${{ github.action_path }}/build-dependencies-dockerfile-logs.sh $f
+          done
+        else
+          # from setup.cfg...
+          export GITHUB_ACTION_PATH=${{ github.action_path }}
+          ${{ github.action_path }}/build-dependencies-logs.sh
+        fi
```

Comment:
We might want to do both?  Maybe an actions option would be good for this.

Reply:
I want to make this as automatic as possible. Packages that do both act as libraries and applications, which I'm not sure is a great pattern. But yeah, it does happen... I assume that if Dockerfile(s) are present, that's the package's intended use

Reply:
I suppose a mono-repo would have a compelling case for acting as a library and application

Reply:
_Thinks of my krs repo, which does both_

It's a bit of an oddball, that's for sure.  I'm not sure I like it, but it is what it is.

Reply:
What's a good name for the option? `all_dependencies_logs`, `local_and_dockerfile`, `app_and_api`, ...

Reply:
update: punting issue and seeing if it surfaces as an issue/enhancement in the future